internal/datanode/data_node.go

```diff
@@ -1074,16 +1074,19 @@ func (node *DataNode) Import(ctx context.Context, req *datapb.ImportTaskRequest)
 
 	// parse files and generate segments
 	segmentSize := int64(Params.DataCoordCfg.SegmentMaxSize) * 1024 * 1024
-	importWrapper := importutil.NewImportWrapper(newCtx, colInfo.GetSchema(), colInfo.GetShardsNum(), segmentSize, node.rowIDAllocator, node.chunkManager,
-		importFlushReqFunc(node, req, importResult, colInfo.GetSchema(), ts), importResult, reportFunc)
+	importWrapper := importutil.NewImportWrapper(newCtx, colInfo.GetSchema(), colInfo.GetShardsNum(), segmentSize, node.rowIDAllocator,
+		node.chunkManager, importResult, reportFunc)
+	importWrapper.SetCallbackFunctions(assignSegmentFunc(node, req),
```

Comment:
Why not put them in NewImportWrapper?

Reply:
too many parameters.

Reply:
True but a broker should do a better job. Having a separate callback setter could be error-prone.

Not super important tho.